src/wwwroot/startupscripts/startagent-win.cmd

```diff
@@ -29,6 +30,11 @@ copy /Y %HELIX_WORKITEM_PAYLOAD%\.credentials %WORKSPACEPATH%
 call %WORKSPACEPATH%\run.cmd
 
 set LASTEXITCODE=%errorlevel%
+
+REM At this point the agent job is done.  If we had a non-delete-able workspace,
```

Comment:
How are we deciding whether the workspace is non-deletable?

Reply:
These three lines:
```
rmdir /S /Q %WORKSPACEPATH%
...
IF NOT EXIST "%WORKSPACEPATH%" goto :CREATE_WORKSPACE
set reboot_required=true
```
... specifically if it failed to delete it in the first line, only then will it ever run `set reboot_required=true`

Reply:
Got it, was reading it wrong, lgtm